bird_classify.py

```diff
@@ -101,43 +103,74 @@ def main():
     logging.basicConfig(filename='%s/results.log'%storage_dir,
                         format='%(asctime)s-%(message)s',
                         level=logging.DEBUG)
-
     last_time = time.monotonic()
     last_results = [('label', 0)]
-    def user_callback(image,svg_canvas):
+
+    def user_callback(image, svg_canvas):
         nonlocal last_time
         nonlocal last_results
         start_time = time.monotonic()
-        results = engine.classify_with_image(image, threshold=args.threshold, top_k=args.top_k)
+        input_tensor_shape = interpreter.get_input_details()[0]['shape']
+        if (input_tensor_shape.size != 4 or
+                input_tensor_shape[0] != 1):
+            raise RuntimeError(
+                'Invalid input tensor shape! Expected: [1, height, width, channel]')
+        _, height, width, channel = input_tensor_shape
+        img = image.resize((width, height), Image.NEAREST)
+        # Handle color space conversion.
+        if channel == 1:
+            img = img.convert('L')
+        elif channel == 3:
+            img = img.convert('RGB')
+        else:
+            raise ValueError(
+                'Invalid input tensor channel! Expected: 1 or 3. Actual: %d' % channel)
+
+        input_tensor = np.asarray(img).flatten()
+        if args.top_k <= 0:
+            raise ValueError('top_k must be positive!')
+        run_inference(interpreter, input_tensor)
+        results = get_classes(interpreter, args.top_k, args.threshold)
         end_time = time.monotonic()
+        play_sounds = [labels[i] for i, score in results]
         results = [(labels[i], score) for i, score in results]
-
+        # play_sounds.append('fox squirrel, eastern fox squirrel, Sciurus niger')
         if args.print:
-          print_results(start_time,last_time, end_time, results)
+            print_results(start_time,last_time, end_time, results)
 
-        if args.training:
-          if do_training(results,last_results,args.top_k):
-            save_data(image,results, storage_dir)
+        if args.training == 'True':
+            if do_training(results,last_results,args.top_k):
+                save_data(image,results, storage_dir)
         else:
-          #Custom model mode:
-          #The labels can be modified to detect/deter user-selected items
-          if results[0][0] !='background':
-            save_data(image, storage_dir,results)
-          if 'fox squirrel, eastern fox squirrel, Sciurus niger' in results:
-            playsound(args.sound)
-            logging.info('Deterrent sounded')
-
-        last_results=results
+            # Custom model mode:
+            # The labels can be modified to detect/deter user-selected items
+            if len(results) > 0 :
+                if results[0][0] != 'background':
+                    save_data(image,  results, storage_dir)
+
+        if args.training == 'False':
+            if 'fox squirrel, eastern fox squirrel, Sciurus niger' in play_sounds:
```

Comment:
I guess it's better to define a string constant, e.g. something like `FOX_SQUIREL_LABEL`.